sdk/include/platform/sunburst/platform-spi.hh

```diff
@@ -0,0 +1,175 @@
+		ConfigurationMSBFirst = 1u << 29,
+		/// The phase of the spi_clk signal. When CPHA is 0 data is sampled on
+		/// the leading edge and changes on the trailing edge. The first data
+		/// bit is immediately available before the first leading edge of the
+		/// clock when transmission begins. When CPHA is 1 data is sampled on
+		/// the trailing edge and change on the leading edge.
+		ConfigurationCPHA = 1u << 30,
+		/// The polarity of the spi_clk signal. When CPOL is 0 clock is low when
+		/// idle and the leading edge is positive. When CPOL is 1 clock is high
+		/// when idle and the leading edge is negative
+		ConfigurationCPOL = 1u << 31,
+	};
+
+	/// Control Register Fields
+	enum : uint32_t
+	{
+		/// Write 1 to clear the transmit FIFO.
+		ControlTxClear = 1 << 0,
+		/// Write 1 to clear the receive FIFO.
+		ControlRxClear = 1 << 1,
+		/// When set bytes from the transmit FIFO are sent. When clear the state
+		/// of the outgoing spi_cipo is undefined whilst the SPI clock is
+		/// running.
+		ControlTxEnable = 1 << 2,
+		/// When set incoming bits are written to the receive FIFO. When clear
+		/// incoming bits are ignored.
+		ControlRxEnable = 1 << 3,
+		/// The watermark level for the transmit FIFO, depending on the value
+		/// the interrupt will trigger at different points
+		ControlTxWatermarkMask = 0xf << 4,
+		/// The watermark level for the receive FIFO, depending on the value the
+		/// interrupt will trigger at different points
+		ControlRxWatermarkMask = 0xf << 8,
+	};
+
+	/// Status Register Fields
+	enum : uint32_t
+	{
+		/// Number of items in the transmit FIFO.
+		StatusTxFifoLevel = 0xffu << 0,
+		/// Number of items in the receive FIFO.
+		StatusRxFifoLevel = 0xffu << 8,
+		/// When set the transmit FIFO is full and any data written to it will
+		/// be ignored.
+		StatusTxFifoFull = 1u << 16,
+		/// When set the receive FIFO is empty and any data read from it will be
+		/// undefined.
+		StatusRxFifoEmpty = 1u << 17,
+		/// When set the SPI block is idle and can accept a new start command.
+		StatusIdle = 1u << 18,
+	};
+
+	/// Start Register Fields
+	enum : uint32_t
+	{
+		/// Number of bytes to receive/transmit in the SPI operation
+		StartByteCountMask = 0x7ffu,
+	};
+
+	/// Initialises the SPI block
+	void init(const bool     ClkPolarity,
+	          const bool     ClkPhase,
+	          const bool     MsbFirst,
+	          const uint16_t HalfClkPeriod) volatile
+	{
+		configuration = (ClkPolarity ? 1 << 31 : 0) | (ClkPhase ? 1 << 30 : 0) |
+		                (MsbFirst ? 1 << 29 : 0) | HalfClkPeriod;
+	}
+
+	/// Waits for the SPI device to become idle
+	void wait_idle() volatile
+	{
+		// Wait whilst IDLE field in STATUS is low
+		while ((status & StatusIdle) == 0) {}
+	}
+
+	/// Sends `Len` bytes from the given `data` buffer,
+	/// where `Len` is at most `0x7ff`.
+	void transmit(const uint8_t data[], const uint16_t Len) volatile
+	{
+		const uint32_t LenTruncated = Len & StartByteCountMask;
```

Comment:
Maybe a `Debug::Assert` that `Len <= 0x7ff`?

Reply:
Ah yes, good call